### PR TITLE
[WIP] RepoTree: alternate design

### PR DESCRIPTION
**Pull request by skshetry**
* [ ] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [ ] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

Yet Another Tree Abstraction :cry:

### CODE REVIEW

File: dvc/repo/ls.py [hunk 1]
```diff
@@ -76,9 +76,10 @@ def onerror(exc):
             if not recursive:
                 for dname in dirs:
                     info = PathInfo(root) / dname
-                    if not dvc_only or (
-                        tree.dvctree and tree.dvctree.exists(info)
-                    ):
+                    t = tree._get_tree(  # pylint: disable=protected-access
+                        info
```
Comment (skshetry) [inline dvc/repo/ls.py:80]: This is of-course going to be fixed by `stats()` as I said in other PR.

**Comment by efiop** [comment]
No longer needed, since https://github.com/iterative/dvc/pull/4381 is merged.

**Comment by skshetry** [comment]
@efiop, I thought you'd like this? Anyway, thanks for merging. :)

**Comment by efiop** [comment]
@skshetry I like it, but the other PR is more wholesome :) I think we'll definitely introduce some good ideas from this PR in the upcoming changes :slightly_smiling_face:

**Pull request closed without merging**